### PR TITLE
Fix bug: missing return parameter for connectionless CTE receive enable

### LINKED ISSUE #35781: Missing response parameter for HCI_LE_Set_Connectionless_IQ_Sampling_Enable HCI command
**Describe the bug**
Response HCI_Command_Complete event generated by BLE controller on HCI_LE_Set_Connectionless_IQ_Sampling_Enable command has missing return parameter: `sync_handle`

**To Reproduce**
Two nRF52833 DKs are required. 
First one must run direction_finding_connectionless_tx sample.
Second one must run direction_finding_connectionless_rx sample.

Error may not be observed when BLE controller runs with Zephyrs host. 
It may be observer if controller is working with host that expect correct `sync_handle` return parameter.
In such situation every call to enable CTE receive and sampling in controller should fail due to:
- expected different length of response event,
- wrong `sync_handle` parameter value. 

**Expected behavior**
Controller should include `sync_handle` parameter in returned HCI_Command_Complete event.
Host implementation should fail with `-EIO` if the `sync_handle` parameter is invalid.

**Impact**
It may impact all BLE host implementations that expect and verify correct return parameters in HCI_Command_Complete for HCI_LE_Set_Connectionless_IQ_Sampling_Enable.
It does not affect Zephyrs host implementation because the parameter was not verified there. 

**Logs and console output**

**Environment (please complete the following information):**
 - Current main branch: e.g commit: 94e8c2eb81f390a9226a3b70fdb69a002642f274.

**Additional context**

### PR DESCRIPTION
**Pull request by ppryga-nordic**
Fixes #35781

Add fixes for 
- controller, to send correct event parameters as a return values for HCI_LE_Set_Connectionless_IQ_Sampling_Enable
- host, to check if `sync_handle` value is correct.

Adds missing return parameters declaration: `struct bt_hci_rp_le_set_cl_cte_sampling_enable`.

### CODE REVIEW

File: subsys/bluetooth/controller/hci/hci.c [hunk 1]
```diff
@@ -2602,7 +2603,10 @@ static void le_df_set_cl_iq_sampling_enable(struct net_buf *buf, struct net_buf
 						 cmd->switch_pattern_len,
 						 cmd->ant_ids);
 
-	*evt = cmd_complete_status(status);
+	rp = hci_cmd_complete(evt, sizeof(*rp));
+
+	rp->status = status;
+	rp->sync_handle = sync_handle;
```
Comment (jhedberg) [inline subsys/bluetooth/controller/hci/hci.c:2609]: `cpu_to_le16(sync_handle)`?

**Comment by ppryga** [comment]
@carlescufi what do you think, would it be good to include this fix into 2.6 release?

File: subsys/bluetooth/controller/hci/hci.c (hunk 1 above)
Comment (ppryga) [reply subsys/bluetooth/controller/hci/hci.c:2609]: Ah, thanks! I'll change it to `rp->sync_handle = cmd->sync_handle;`

Comment (jhedberg) [reply subsys/bluetooth/controller/hci/hci.c:2609]: I'm not sure how the command-response buffer handling is today, you might want to be a bit careful with that if the command buffer gets reused for the event. In that case you might risk corruption if e.g. rp->status occupies the same memory as cmd->sync_handle. @carlescufi or @cvinayak should know that's safe to do in the controller.

Comment (ppryga) [reply subsys/bluetooth/controller/hci/hci.c]: Corrected

File: subsys/bluetooth/controller/hci/hci.c [hunk 2]
```diff
@@ -2590,10 +2590,12 @@ static void le_df_set_cl_cte_enable(struct net_buf *buf, struct net_buf **evt)
 static void le_df_set_cl_iq_sampling_enable(struct net_buf *buf, struct net_buf **evt)
 {
 	struct bt_hci_cp_le_set_cl_cte_sampling_enable *cmd = (void *)buf->data;
-	uint16_t sync_handle;
+	struct bt_hci_rp_le_set_cl_cte_sampling_enable *rp;
+	uint16_t sync_handle, sync_handle_le16;
 	uint8_t status;
 
-	sync_handle = sys_le16_to_cpu(cmd->sync_handle);
+	sync_handle_le16 = cmd->sync_handle;
```
Comment (joerchan) [inline subsys/bluetooth/controller/hci/hci.c:2597]: I'd skip this optimization and just convert it back again when you set it.
On ARM devices this is a no-op anyway.

File: subsys/bluetooth/host/direction.c [hunk 3]
```diff
@@ -300,14 +301,23 @@ static int hci_df_set_cl_cte_rx_enable(struct bt_le_per_adv_sync *sync, bool ena
...
 	}
 
-	sync->cte_type = (enable ? params->cte_type : 0);
+	rp = (void *)rsp->data;
+	if (sync->handle != sys_le16_to_cpu(rp->sync_handle)) {
+		err = -EIO;
+	}
 
-	return 0;
+	if (!err) {
```
Comment (joerchan) [inline subsys/bluetooth/host/direction.c:314]: Better as an else?

File: subsys/bluetooth/controller/hci/hci.c (hunk 1 above)
Comment (joerchan) [reply subsys/bluetooth/controller/hci/hci.c]: In a combined build the command complete event would be the same buffer as the command. `bt_buf_get_cmd_complete` in `subsys/bluetooth/host/buf.c`.

Comment (ppryga) [reply subsys/bluetooth/controller/hci/hci.c]: I've figured it out when I run test code. Changed according to @joerchan suggestion https://github.com/zephyrproject-rtos/zephyr/pull/35788#discussion_r641742811

File: subsys/bluetooth/controller/hci/hci.c (hunk 2 above)
Comment (ppryga) [reply subsys/bluetooth/controller/hci/hci.c]: You are right, assembly generated is the same with or without additional variable. Changes as suggested.

File: subsys/bluetooth/host/direction.c (hunk 3 above)
Comment (ppryga) [reply subsys/bluetooth/host/direction.c]: Changed to else

**Comment by ppryga** [comment]
@joerchan and @jhedberg requested changes were implemented.

**Review by jhedberg — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Pull request merged**